pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -305,13 +305,19 @@ def ddp_train(self, process_idx, model):
         :param cluster_obj:
         :return:
         """
-        # node rank using relative slurm id if under slurm management
+        # if under slurm management, use SLURM_NODEID,
+        # if torchelastic, use GROUP_RANK,
         # otherwise use given node rank or default to node rank 0
-        try:
-            node_id = os.environ['SLURM_NODEID'] if self.is_slurm_managing_tasks else os.environ['NODE_RANK']
+        node_keys = ['SLURM_NODEID', 'NODE_RANK', 'GROUP_RANK']
```

Comment:
I'm concerned about cases when multiple of these are set. Can we log a warning if we use one of these IDs and the other two are set? I think writing the code that way would make it easier to understand what's happening here too.

Reply:
good idea.